### PR TITLE
MELEE_DAMAGE and ITEM_DAMAGE_BASH enchantments should affect bashing damage to tile

### PR DESCRIPTION
**Pull request by fyhannes**
#### Summary
Bugfixes "Smashing now incorporates any MELEE_DAMAGE and STRENGTH enchantments"

#### Purpose of change
#66434. STRENGTH and ITEM_DAMAGE_BASH enchantments should influence SMASH capabilities but currently only takes in arm_strength and the weapon's bash damage.

#### Describe the solution
Use `calculate_by_enchantment` to help contribute to the amount of smash damage done.

#### Describe alternatives you've considered
I initially thought about just linearly adding the STRENGTH and MELEE_DAMAGE bonuses to the smash damage, but I figured that `calculate_by_enchantment` was better suited to handle the smash damage calculation.  

#### Testing
1. Create a world with Magiclysm mod. Set character with strength of 4.
2. Locate retirement home and teleport to outside. There should be reinforced glass where the home faces the retirement home  parking lot.
3. Debug spawn pipe_mace and add TACTILE_KINESIS
4. Cast TACTILE_KINESIS and wield pipe mace
5. ~Terrorize senior citizens~ Smash reinforced glass until it breaks.

Reinforced glass has a bash_min of 40 and a pipe_mace has a bash rating of 24, so casting TACTILE KINESIS should allow the player to smash the window with a STRENGTH of 4

#### Additional context
Screenshot 2023-07-07 at 1 02 44 PM
Here's an example of reinforced glass breaking. My character has 4 strength, 4 dexterity, and has the GRAVITY_CONTROL effect. My character is also wielding a pipe mace and generic survivor clothing.

### CODE REVIEW

**Comment by GuardianDll** [comment]
Strength doesn't influence it? I thought it add points to your character strength, which affect the bash output?
Anyway i think ITEM_DAMAGE_BASH should also be added here

**Comment by fyhannes** [comment]
> Strength doesn't influence it? I thought it add points to your character strength, which affect the bash output?
Yeah it does. I'll remove the STRENGTH enchantment calculation I added since STRENGTH enchantments will affect `cur_str` which is used in calculating the bash output. Thanks for the catch.

>Anyway i think ITEM_DAMAGE_BASH should also be added here
Good point on ITEM_DAMAGE_BASH, I'll add that as well.

File: src/handle_action.cpp [hunk 1]
```diff
@@ -741,19 +741,20 @@ static void smash()
...
+        smashskill += mon->mech_str_addition() + mon->type->melee_dice * mon->type->melee_sides;
         mech_smash = true;
-    } else {
-        smashskill = player_character.get_arm_str();
-        if( player_character.get_wielded_item() ) {
-            smashskill += player_character.get_wielded_item()->damage_melee( damage_bash );
-        }
-    }
+    } else if( player_character.get_wielded_item() ) {
+        smashskill += player_character.calculate_by_enchantment(
```
Comment (bombasticSlacks) [inline src/handle_action.cpp:751]: Why only if the player is weilding a weapon?

Comment (fyhannes) [reply src/handle_action.cpp:751]: If you mean "why do I only factor in MELEE_DAMAGE enchantment if the player is wielding a weapon", then you're right, this is a mistake. I should be linearly adding smashskill by the bash rating of the weapon they're currently holding and then perform the enchantment calculation afterwards.

I think there are two ways I could do it from there: the first is to call 

```
smashskill = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::MELEE_DAMAGE,
                  true );
smashskill = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::ITEM_DAMAGE_BASH,
                  true );
```

or, I could do something like

```
int melee_damage_enchantment =  player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::MELEE_DAMAGE,
                  true );
int item_damage_bash_enchantment = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::ITEM_DAMAGE_BASH,
                  true );
smashskill += melee_damage_enchantment + item_damage_bash_enchantment
```

I'm leaning towards the second one but would like to hear your thoughts.

Comment (bombasticSlacks) [reply src/handle_action.cpp:751]: I think #1 is easier to read 😄

File: src/handle_action.cpp [hunk 2]
```diff
@@ -741,19 +741,20 @@ static void smash()
...
-        if( player_character.get_wielded_item() ) {
-            smashskill += player_character.get_wielded_item()->damage_melee( damage_bash );
-        }
-    }
+    } else if( player_character.get_wielded_item() ) {
+        smashskill += player_character.get_wielded_item()->damage_melee( damage_bash );
+    }
+    smashskill = player_character.calculate_by_enchantment( smashskill, enchant_vals::mod::MELEE_DAMAGE,
+                 true );
+    smashskill = player_character.calculate_by_enchantment( smashskill,
```
Comment (dseguin) [inline src/handle_action.cpp:755]: Is the assignment operator intentional? Looks like it should be a `+=` instead

Comment (fyhannes) [reply src/handle_action.cpp:755]: Yeah, it's intentional. `calculate_by_enchantment`'s first argument is the value we want to modify, and then it returns that value after adding/multiplying enchantment bonuses to it.

Comment (dseguin) [reply src/handle_action.cpp:755]: I see, cheers :+1:

**Pull request merged**